manager/src/main/java/com/redhat/service/smartevents/manager/api/user/validators/processors/ProcessorGatewayConstraintValidator.java

```diff
@@ -48,9 +48,7 @@ public boolean isValid(ProcessorRequest value, ConstraintValidatorContext contex
             return false;
         }
 
-        return action != null
-                ? isValidGateway(action, context, gatewayConfigurator::getActionValidator)
-                : isValidGateway(source, context, gatewayConfigurator::getSourceValidator);
+        return isValidGateway(action != null ? action : source, context);
```

Comment:
we might add a transient property in the DTO called `getPayload()` that does this if? so we don't have to spread it around the code